odmantic/engine.py

```diff
@@ -511,28 +551,37 @@ async def find_one(
             return None
         return results[0]
 
+    async def _save_collection_updates(
+        self,
+        collection_updates: CollectionUpdatesType,
+        session: "AsyncIOMotorClientSession",
+    ) -> None:
+        # reverse so that the last collections added, the ones for sub-documents, are
+        # saved first
+        for collection_name, updates in reversed(list(collection_updates.items())):
+            collection = self._get_collection_from_name(collection_name)
+            update_operations = [update[0] for update in updates]
+            update_instances = [update[1] for update in updates]
```

Comment:
`[update[0] for update in updates]` is a bit unusual to see for me.

Maybe this would be a bit clearer
```python
update_operations, update_instances = zip(*updates)
```

But still quite strange, maybe the easiest way would be to build a small dataclass or a namedtuple to have something a bit more explicit.

Reply:
Sure thing! Done ✔️